### ISSUE TITLE
Improve parser error messages

### ISSUE THREAD

**Issue by chrarnold** [snakemake/snakemake#1729]
**Snakemake version**
all

**Describe the bug**
I receive the following error upon starting Snakemake for a Snakefile:
`TokenError:
('EOF in multi-line statement', (257, 0))
`
Very hard to debug, not even sure in which file to look: the Snakefile, the config file? Is 257 the line number? Would be great if such errors could be improved so debugging becomes easier. The end of the files looks ok, I dont see anything obvious.

**Comment by dariober** [comment]
I doubt that `TokenError ...` is the *only* output you got from executing snakemake. Most likely there is something before and after that you are not showing and that should help debugging. You may have a point that the error message could be improved but from what you show it's impossible to tell, at least for me...

**Comment by chrarnold** [comment]
No, that was indeed all, therefore so hard to debug. I always post the full error message... it turned out the reason was a missing closing parenthesis in one of the lines above the rule section with custom Python code.

**Comment by dariober** [comment]
I think you can close this issue since `TokenError` comes from python not snakemake so I think there is not much the snakemake developers can do to improve the message.
